service/dap/server.go

```diff
@@ -354,24 +354,15 @@ func (s *Server) onLaunchRequest(request *dap.LaunchRequest) {
 	}
 
 	stop, ok := request.Arguments["stopOnEntry"]
-	s.stopOnEntry = (ok && stop == true)
+	s.stopOnEntry = ok && stop == true
 
 	// TODO(polina): support target args
 	s.config.ProcessArgs = []string{program}
-	s.config.WorkingDir = filepath.Dir(program)
-
-	config := &debugger.Config{
-		WorkingDir:           s.config.WorkingDir,
-		AttachPid:            0,
-		CoreFile:             "",
-		Backend:              s.config.Backend,
-		Foreground:           s.config.Foreground,
-		DebugInfoDirectories: s.config.DebugInfoDirectories,
-		CheckGoVersion:       s.config.CheckGoVersion,
-		TTY:                  s.config.TTY,
-	}
+	s.config.Debugger.WorkingDir = filepath.Dir(program)
+
+	config := s.config.Debugger
 	var err error
-	if s.debugger, err = debugger.New(config, s.config.ProcessArgs); err != nil {
+	if s.debugger, err = debugger.New(&config, s.config.ProcessArgs); err != nil {
```

Comment:
En, maybe we can remove temporary variable `config` and 
```
debugger.New(&s.config.Debugger, s.config.ProcessArgs)
```
is enough.

Reply:
True, that would work. I did this mostly so that the debugger object got a pointer to a copy of the config and not the exact config embedded in `s.config`. Not a big deal either way, I suppose.